components/script/dom/window.rs

```diff
@@ -1227,14 +1169,6 @@ impl Window {
         self.paint_worklet.or_init(|| self.new_paint_worklet())
     }
 
-    pub fn task_canceller(&self, name: TaskSourceName) -> TaskCanceller {
-        let mut flags = self.ignore_further_async_events.borrow_mut();
```

Comment:
The `ignore_further_async_events` field of `Window` is still used in a few places, such as `ignore_all_events`, this makes the task cancelling inside `ignore_all_events` not have effect on tasks that have been scheduled via the new `TaskManager`, since the `Arc<AtomicBool>` are not shared between `ignore_further_async_events` and `TaskManager.task_cancellers`. 

So we need to remove `ignore_further_async_events` from `Window`, and make methods like `Window.ignore_all_events` used the `TaskManager` instead.

Reply:
Wow, there's no way I'd have seen that. Thank you so much.

One question, before I implement that, is there an (easy) way to properly fix this? I.e. no failure at all. Or would it fall outside the scope of this pr?

Reply:
You're welcome.

I'm not sure if there is an easy way to fix the intermittent failure, although once the above is fixed in one commit, you could try in a separate commit the suggestion I made at https://github.com/servo/servo/pull/21804#issuecomment-435942965 (although since it is intermittent, you might have a hard time actually knowing for sure whether it helped or not).